numbers-csharp/ReverseNumberAsInteger/ReverseNumberAsInteger/ReverseNumbers.cs

```diff
@@ -0,0 +1,155 @@
+﻿namespace ReverseNumberAsInteger
+{
+    public class ReverseNumbers
+    {
+        readonly static int maxValueDiv10 = int.MaxValue / 10;
+        readonly static int minValueDiv10 = int.MinValue / 10;
+
+        public static int ReverseUsingDigitExtractionAndReconstruction(int num)
+        {
+            if (num == int.MaxValue || num == int.MinValue || num == 0)
+            {
+                return 0;
+            }
+
+            int reversedNumber = 0;
+
+            while (num != 0)
+            {
+                int remainder = num % 10;
+
+                if (reversedNumber > maxValueDiv10 || reversedNumber < minValueDiv10)
+                {
+                    return 0;
+                }
+
+                reversedNumber = reversedNumber * 10 + remainder;
+                num /= 10;
+            }
+
+            return reversedNumber;
+        }
+
+        public static int ReverseUsingRecursion(int num, int reversedNumber = 0)
+        {
+            if (num == int.MaxValue || num == int.MinValue || num == 0)
+            {
+                return reversedNumber;
+            }
+
+            if (reversedNumber > maxValueDiv10 || reversedNumber < minValueDiv10)
+            {
+                return 0;
+            }
+
+            int remainder = num % 10;
+
+            reversedNumber = reversedNumber * 10 + remainder;
+
+            return ReverseUsingRecursion(num / 10, reversedNumber);
+        }
+
+        public static int ReverseUsingMathPow(int num)
+        {
+            if (num == int.MaxValue || num == int.MinValue || num == 0)
+            {
+                return 0;
+            }
+
+            bool isNegative = num < 0;
+            num = isNegative ? -num : num;
+
+            int length = (int)Math.Floor(Math.Log10(num) + 1);            
+            int powerOf10 = (int)Math.Pow(10, length - 1);
+            int reversedNumber = 0;
+
+            for (int i = 1; i <= length; i++)
```

Comment:
we should index from 0 to < length, rather than from 1 to length. less than typically is more optimized than less than equal (it may end up the same after the compiler does some tweaking, but less than is generally preferred.